applications/RANSApplication/custom_elements/convection_diffusion_reaction_residual_based_flux_corrected_element.h

```diff
@@ -0,0 +1,950 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_CONVECTION_DIFFUSION_REACTION_RESIDUAL_BASED_FLUX_CORRECTED_ELEMENT_H_INCLUDED)
+#define KRATOS_CONVECTION_DIFFUSION_REACTION_RESIDUAL_BASED_FLUX_CORRECTED_ELEMENT_H_INCLUDED
+
+// System includes
+#include <cmath>
+
+// External includes
+
+// Project includes
+#include "includes/cfd_variables.h"
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/element.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/time_discretization.h"
+
+// Application includes
+#include "convection_diffusion_reaction_stabilization_utilities.h"
+#include "custom_utilities/rans_calculation_utilities.h"
+#include "rans_application_variables.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+template <
+    IndexType TDim,
+    IndexType TNumNodes,
+    class TConvectionDiffusionReactionData>
+class ConvectionDiffusionReactionResidualBasedFluxCorrectedElement : public Element
```

Comment:
Shouldn't this derive from the `ConvectionDiffusionReacion` element?

Reply:
done :)